include/lib_state_machine.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif // __cplusplus
```

Comment:
Do you think we will need this if it is all self-contained in `bm_common`/`bm_core` or are there still aspects in `bm_protocol` that utilize this file? If so is it best to keep the `extern "C"` directive in here?